File: javascript/2125. Number of Laser Beams in a Bank.js
/**
 * @param {string[]} bank
 * @return {number}
 */
var numberOfBeams = function (bank) {
  var prevBeans = 0;
  var ans = 0;
  for (let str of bank) {
    var noCurBeam = getNoBeams(str);
    if (noCurBeam === 0) {
      continue;
    } else if (prevBeans === 0) {
      prevBeans = noCurBeam;
    } else {
      ans += prevBeans * noCurBeam;
      prevBeans = noCurBeam;
    }
  }
  return ans;
};

const getNoBeams = (row) => {
  var ans = 0;
  for (let i = 0; i < row.length; i++) {
    if (row.charAt(i) === "1") {
      ans += 1;
    }
  }
  return ans;
};

File: java/2125. Number of Laser Beams in a Bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int prevRowBeams = 0;
        int ans = 0;
        for(String row: bank){
            int curRowBeams = getNumberOfBeams(row);
            if(curRowBeams == 0){
                continue;
            }
            else if(prevRowBeams == 0){
                prevRowBeams = curRowBeams;
            }
            else{
                ans += (prevRowBeams * curRowBeams);
                prevRowBeams = curRowBeams;
            }
        }
        return ans;
    }

    public int getNumberOfBeams(String bankRow){
        int ans = 0;
        for(int i = 0; i < bankRow.length() ; i++){
            if(bankRow.charAt(i) == '1'){
                ans++;
            }
        }
        return ans;
    }
}